### PR TITLE
add lazarus-3.0 and libqt6pas

### PR DESCRIPTION
**Pull request by dpronin**

### CODE REVIEW

**Comment by dpronin** [comment]
@Amynka @corvus1 ping. Consider merging in these new packages

**Comment by Amynka** [comment]
Will check it and test it.. Thank you

**Comment by dpronin** [comment]
@Amynka I have tested lazarus-3.0 with FPC-3.2.2 and QT6 and it works well. [Here|https://github.com/reyandme/kam_remake/pulls] there are many PRs created with support of the IDE

**Comment by Amynka** [comment]
Alright.. thank you!

**Comment by thesamesam** [comment]
Could you fix the commit message format wrt https://www.gentoo.org/glep/glep-0066.html#commit-messages? Thanks!

**Comment by dpronin** [comment]
@thesamesam check it out

**Comment by dpronin** [comment]
@Amynka @thesamesam merge it in, please. There are not harmful and dangerous changes

**Comment by Amynka** [comment]
I cannot merge it exactly how it is.. I have to make the changes..

File: dev-libs/libqt6pas/metadata.xml
Comment (Amynka) [inline dev-libs/libqt6pas/metadata.xml:9]: Can you remove voron1 as maintainer, since he didnt agree to take this package..

**Review by Amynka — CHANGES_REQUESTED** [review]

File: dev-libs/libqt6pas/metadata.xml
Comment (dpronin) [reply dev-libs/libqt6pas/metadata.xml:9]: Should I just remove the XML-node?

**Comment by dpronin** [comment]
@Amynka done

**Comment by Amynka** [comment]
Also some pkgcheck issues:
  RequiredUseDefaults: version 3.0: profile: 'default/linux/amd64/17.0/musl' (36 total) failed REQUIRED_USE: exactly-one-of ( gtk2 gtk qt5 qt6 )
  UnquotedVariable: version 3.0: unquoted variable FILESDIR on line: 52

**Comment by dpronin** [comment]
@Amynka I have corrected unquoted variable, but I cannot understand what is **RequiredUseDefaults: version 3.0: profile: 'default/linux/amd64/17.0/musl' (36 total) failed REQUIRED_USE: exactly-one-of ( gtk2 gtk qt5 qt6 )**

**Comment by dpronin** [comment]
@Amynka I seem to have fixed all QA warnings of dev-lang/lazarus

**Comment by Amynka** [comment]
> @Amynka I seem to have fixed all QA warnings of dev-lang/lazarus

I wouldnt say its fixed.. 

  RequiredUseDefaults: version 3.0: profile: 'default/linux/amd64/17.0/musl' (36 total) failed REQUIRED_USE: exactly-one-of ( gtk2 gtk qt5 qt6 )

You changed the depend ?gui the useflags.. The complain is about that only one of those can be used as was with the old ebuild either was gtk2 or not gtk2 with libqt5pas. You can also run yourself the `pkgcheck scan` to see the results..

**Comment by dpronin** [comment]
> > @Amynka I seem to have fixed all QA warnings of dev-lang/lazarus
> 
> I wouldnt say its fixed..
> 
> RequiredUseDefaults: version 3.0: profile: 'default/linux/amd64/17.0/musl' (36 total) failed REQUIRED_USE: exactly-one-of ( gtk2 gtk qt5 qt6 )
> 
> You changed the depend ?gui the useflags.. The complain is about that only one of those can be used as was with the old ebuild either was gtk2 or not gtk2 with libqt5pas. You can also run yourself the `pkgcheck scan` to see the results..

So, I still cannot understand what is wrong in sense of changes. Why are they treated as illegal?

**Comment by dpronin** [comment]
@Amynka ping

**Comment by Amynka** [comment]
> @Amynka ping

Did you also check the previous issue?

**Comment by dpronin** [comment]
@Amynka gentoo repository reports that all checks have passed

**Comment by Amynka** [comment]
@dpronin Are you sure about that? I can see it here https://qa-reports.gentoo.org/output/gentoo-ci/1b231f48fb/output.html#dev-lang/lazarus

**Comment by thesamesam** [comment]
I was very much hoping Amy would handle it but I guess that's not going to happen. I'll take a look.

**Comment by dpronin** [comment]
@thesamesam you really a hero saving different situations

File: dev-lang/lazarus/lazarus-3.0.ebuild
Comment (thesamesam) [inline dev-lang/lazarus/lazarus-3.0.ebuild:20]: In a followup (as you've waited long enough), could you make an executive decision as to which framework is preferred? e.g. whatever upstream prefer / what works best?

See https://wiki.gentoo.org/wiki/Project:Qt/Policies https://projects.gentoo.org/qa/policy-guide/use-flags.html#pg0802.

So, for example, if qt6 is best for Lazarus (preferred/default upstream), you would do...
```
IUSE="+gui gtk2 qtk qt5 extras"
REQUIRED_USE="extras? ( gui )"
```

I'm undecided on the REQUIRED_USE for the rest of it and think it should probably go as it doesn't let users easily do "USE=gui qt6" in make.conf without hitting REQUIRED_USE conflicts.

Comment (dpronin) [reply dev-lang/lazarus/lazarus-3.0.ebuild:20]: you meant 
```
IUSE="+gui gtk2 qtk **qt6** extras"
```

Comment (thesamesam) [reply dev-lang/lazarus/lazarus-3.0.ebuild:20]: No - the idea is that you make one of them controlled by USE=gui.

Comment (dpronin) [reply dev-lang/lazarus/lazarus-3.0.ebuild:20]: To conclude, if we prefer qt6 we leave qt6 in IUSE

Comment (dpronin) [reply dev-lang/lazarus/lazarus-3.0.ebuild:20]: @thesamesam see [PR](https://github.com/gentoo/gentoo/pull/37103). Is it the thing you have meant?

**Pull request closed without merging**
